src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyNamesTests.cs

```diff
@@ -0,0 +1,387 @@
+                    public C1(int firstIField, object secondIField)
+                    {
+                        this.firstField = firstIField;
+                        this.secondField = secondIField;
+                    }
+                }",
+            CA1071CSharpFieldResultAt(11, 35, "C1", "firstIField", "firstField"),
+            CA1071CSharpFieldResultAt(11, 55, "C1", "secondIField", "secondField"));
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsDoNotMatchNotJsonCtor_NoDiagnostics_CSharp()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    public int firstField;
+                    public object secondField;
+
+                    public C1(int firstIField, object secondIField)
+                    {
+                        this.firstField = firstIField;
+                        this.secondField = secondIField;
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsDoNotMatchNotJsonCtor_NoDiagnostics_Basic()
+        {
+            await VerifyBasicAnalyzerAsync(@"
+Imports System.Text.Json.Serialization
+
+Public Class C1
+    Public firstField as Integer
+    Public secondField as Object
+
+    Public Sub New(firstIField as Integer, secondIField as Object)
+        Me.firstField = firstIField
+        Me.secondField = secondIField
+    End Sub
+End Class");
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsMatch_NoDiagnostics_CSharp()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+                using System.Text.Json.Serialization;
+
+                public class C1
+                {
+                    public int firstField;
+                    public object secondField;
+
+                    [JsonConstructor]
+                    public C1(int firstField, object secondField)
+                    {
+                        this.firstField = firstField;
+                        this.secondField = secondField;
+                    }
+                }"
+            );
+        }
+
+        [Fact]
+        public async Task CA1071_ClassFieldsMatch_NoDiagnostics_Basic()
+        {
+            await VerifyBasicAnalyzerAsync(@"
+Imports System.Text.Json.Serialization
+
+Public Class C1
+    Public firstField as Integer
+    Public secondField as Object
+
+    <JsonConstructor>
+    Public Sub New(firstField as Integer, secondField as Object)
+        Me.firstField = firstField
+        Me.secondField = secondField
+    End Sub
+End Class");
+        }
+
+        private static async Task VerifyCSharpAnalyzerAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithSystemTextJson50,
```

Comment:
Good point. Seems like it may have an even lesser impact on tests execution time because there won't be an additional metadata reference that has to be resolved/downloaded. Done.